core/services/workflows/engine.go

```diff
@@ -918,6 +920,13 @@ func (e *Engine) executeStep(ctx context.Context, lggr logger.Logger, msg stepRe
 		return nil, nil, err
 	}
 
+	allowed := e.ratelimiter.Allow(e.workflow.owner)
```

Comment:
This is in the wrong place I think -- we should rate limit executions, not rate limit steps; if we've decided an execution can go ahead, we shouldn't cause steps to fail 

Note that there's already some natural rate limiting being applied at the workflow level on steps deu to the fact that there's a worker pool that services steps.

You probably want to add this here instead: https://github.com/smartcontractkit/chainlink/pull/15939/files#diff-b1ca9fad3006c72cc7e4401eb80d180a603350c0cbcdfc121b90ae88172742cdL728